pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PurgeMinionClusterIntegrationTest.java

```diff
@@ -0,0 +1,380 @@
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+/**
+ * Integration test for minion task of type "PurgeTask"
+ */
+public class PurgeMinionClusterIntegrationTest extends BaseClusterIntegrationTest {
+  private static final String PURGE_FIRST_RUN_TABLE = "myTable1";
+  private static final String PURGE_DELTA_PASSED_TABLE = "myTable2";
+  private static final String PURGE_DELTA_NOT_PASSED_TABLE = "myTable3";
+
+  protected PinotHelixTaskResourceManager _helixTaskResourceManager;
+  protected PinotTaskManager _taskManager;
+  protected PinotHelixResourceManager _pinotHelixResourceManager;
+
+  protected final File _segmentDir1 = new File(_tempDir, "segmentDir1");
+  protected final File _segmentDir2 = new File(_tempDir, "segmentDir2");
+  protected final File _segmentDir3 = new File(_tempDir, "segmentDir3");
+
+  protected final File _tarDir1 = new File(_tempDir, "tarDir1");
+  protected final File _tarDir2 = new File(_tempDir, "tarDir2");
+  protected final File _tarDir3 = new File(_tempDir, "tarDir3");
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir1, _tarDir1,
+        _segmentDir2, _tarDir2, _segmentDir3, _tarDir3);
+
+    // Start the Pinot cluster
+    startZk();
+    startController();
+    startBrokers(1);
+    startServers(1);
+
+    // Create and upload the schema and table config
+    Schema schema = createSchema();
+    addSchema(schema);
+    TableConfig purgeTableConfig =
+        createOfflineTableConfig(PURGE_FIRST_RUN_TABLE, getPurgeTaskConfig());
+    TableConfig purgeDeltaPassedTableConfig =
+        createOfflineTableConfig(PURGE_DELTA_PASSED_TABLE, getPurgeTaskConfig());
+    TableConfig purgeDeltaNotPassedTableConfig =
+        createOfflineTableConfig(PURGE_DELTA_NOT_PASSED_TABLE, getPurgeTaskConfig());
+    addTableConfig(purgeTableConfig);
+    addTableConfig(purgeDeltaPassedTableConfig);
+    addTableConfig(purgeDeltaNotPassedTableConfig);
+
+
+
+
+    // Unpack the Avro files
+    List<File> avroFiles = unpackAvroData(_tempDir);
+
+    // Create and upload segments
+    ClusterIntegrationTestUtils
+        .buildSegmentsFromAvro(avroFiles, purgeTableConfig, schema, 0, _segmentDir1, _tarDir1);
+
+    buildSegmentsFromAvroWithPurgeTime(avroFiles, purgeDeltaPassedTableConfig, schema, 0,
+        _segmentDir2, _tarDir2, String.valueOf(System.currentTimeMillis() - 863660000));
+    buildSegmentsFromAvroWithPurgeTime(avroFiles, purgeDeltaNotPassedTableConfig, schema, 0,
+        _segmentDir3, _tarDir3, String.valueOf(System.currentTimeMillis() - 400000));
+
+    uploadSegments(PURGE_FIRST_RUN_TABLE, _tarDir1);
+    uploadSegments(PURGE_DELTA_PASSED_TABLE, _tarDir2);
+    uploadSegments(PURGE_DELTA_NOT_PASSED_TABLE, _tarDir3);
+
+    // Set up the H2 connection
+    setUpH2Connection(avroFiles);
+
+    // Initialize the query generator
+    setUpQueryGenerator(avroFiles);
+
+    startMinion();
```

Comment:
Let's also setup the `RecordPurgerFactory` and `RecordModifierFactory` to do an end-to-end test. You may refer to `PurgeTaskExecutorTest` on how to set them up

Reply:
Done.